pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -69,6 +48,35 @@ func NewMapFromBuildPlan(build *model.Build) (artifact.Map, error) {
 	return res, nil
 }
 
+func buildTerminals(nodeID strfmt.UUID, lookup map[strfmt.UUID]interface{}, result *[]strfmt.UUID) {
+	targetArtifact, ok := lookup[nodeID].(*model.Artifact)
```

Comment:
You're doing two actions here that can fail:

* lookup[nodeID]
* lookup[nodeID].(*model.Artifact)

I think the conditional is only addressing the first, and if `lookup[nodeID]` does not exist it will probably panic.

Reply:
I don't believe it will panic if there is no entry in the map, the conversion will simply fail. See here: https://go.dev/play/p/I8GmYcNDzQo

Let me know if that addresses your concerns, if not I'm happy to add more checks.

Reply:
Nice, that does indeed address my concern, thanks for being thorough!